www/app/Tools/ComposeTransformTool.php

```
@@ -0,0 +1,415 @@
+        }
+
+        $serviceCommand = $this->getServiceCommand($service);
+        $indent = str_repeat(" ", $serviceIndent + 2);
+
+        // Build the entrypoint with copy commands, running as root for permission to write system dirs
+        $copyScript = implode(" && ", $copyCommands);
+        $lines = [];
+        $lines[] = "{$indent}user: root";
+        $lines[] = "{$indent}entrypoint: [\"/bin/sh\", \"-c\", \"{$copyScript} && {$serviceCommand}\"]";
+        return implode("\n", $lines);
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'injectEntrypoint'. (undefined)

(UnusedPrivateMethod)